tx-pool/src/process/block_template.rs

```diff
@@ -273,19 +274,35 @@ impl Future for BlockTemplateBuilder {
         let (bytes_limit, _, version) = self.args;
         let cycles_limit = consensus.max_block_cycles();
         let uncles_count_limit = consensus.max_uncles_num() as u32;
+        let compact_target: Uint32 = self.current_epoch.compact_target().into();
 
         // Should recalculate current time after create cellbase (create cellbase may spend a lot of time)
         let current_time = cmp::max(unix_time_as_millis(), tip_header.timestamp() + 1);
+
+        let epoch: EpochNumberWithFraction = self
+            .current_epoch
+            .number_with_fraction(candidate_number)
+            .into();
+
+
+        let raw = RawHeader::new_builder()
```

Comment:
This step misses the cellbase transaction, and the raw header is wrong. There's a easy way to get raw header from block template.

```
let block_template = BlockTemplate {
  // omit
};

let block: packed::Block = block_template.into();
```

Reply:
Thank you, that simplifies it a lot.